src/ios/CDVLocation.m

```diff
@@ -80,87 +80,88 @@ - (BOOL)isAuthorized
+            BOOL authStatusAvailable = [CLLocationManager respondsToSelector:@selector(authorizationStatus)]; // iOS 4.2+
+            if (authStatusAvailable) {
+                NSUInteger code = [CLLocationManager authorizationStatus];
+                if (code == kCLAuthorizationStatusNotDetermined) {
+                    // could return POSITION_UNAVAILABLE but need to coordinate with other platforms
+                    message = @"User undecided on application's use of location services.";
+                } else if (code == kCLAuthorizationStatusRestricted) {
+                    message = @"Application's use of location services is restricted.";
+                }
+            }
+            // PERMISSIONDENIED is only PositionError that makes sense when authorization denied
+            [strongSelf returnLocationError:PERMISSIONDENIED withMessage:message];
 
-        return;
-    }
+            return;
+        }
 
-#ifdef __IPHONE_8_0
-    NSUInteger code = [CLLocationManager authorizationStatus];
-    if (code == kCLAuthorizationStatusNotDetermined && ([self.locationManager respondsToSelector:@selector(requestAlwaysAuthorization)] || [self.locationManager respondsToSelector:@selector(requestWhenInUseAuthorization)])) { //iOS8+
-        __highAccuracyEnabled = enableHighAccuracy;
-        if([[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationWhenInUseUsageDescription"]){
-            [self.locationManager requestWhenInUseAuthorization];
-        } else if([[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysUsageDescription"]) {
-            [self.locationManager  requestAlwaysAuthorization];
+        // Tell the location manager to start notifying us of location updates. We
+        // first stop, and then start the updating to ensure we get at least one
+        // update, even if our location did not change.
+        [strongSelf.locationManager stopUpdatingLocation];
+        [strongSelf.locationManager startUpdatingLocation];
+        strongSelf->__locationStarted = YES;
+        if (enableHighAccuracy) {
+            strongSelf->__highAccuracyEnabled = YES;
+            // Set distance filter to 5 for a high accuracy. Setting it to "kCLDistanceFilterNone" could provide a
+            // higher accuracy, but it's also just spamming the callback with useless reports which drain the battery.
+            strongSelf.locationManager.distanceFilter = 5;
+            // Set desired accuracy to Best.
+            strongSelf.locationManager.desiredAccuracy = kCLLocationAccuracyBest;
         } else {
-            NSLog(@"[Warning] No NSLocationAlwaysUsageDescription or NSLocationWhenInUseUsageDescription key is defined in the Info.plist file.");
+            strongSelf->__highAccuracyEnabled = NO;
+            strongSelf.locationManager.distanceFilter = 10;
+            strongSelf.locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers;
         }
-        return;
-    }
-#endif
```

Comment:
As already stated in another comment, this is something that's still used so it shouldn't be deleted.

Reply:
Fixed.